include/rdma/fabric.h

```diff
@@ -779,11 +779,14 @@ struct fi_param {
 int fi_getparams(struct fi_param **params, int *count);
 void fi_freeparams(struct fi_param *params);
 
-/* Dummy definitions for removed flags/caps/types. For compiling old fabtests */
+/* Dummy definitions for removed flags/caps/types. For compiling old fabtests and MPICH */
 #define FI_VARIABLE_MSG		0ULL
 #define FI_NOTIFY_FLAGS_ONLY	0ULL
 #define FI_RESTRICTED_COMP	0ULL
 #define FI_EP_SOCK_STREAM	FI_EP_UNSPEC
+#define FI_ORDER_NONE		0ULL
+#define FI_MR_BASIC		(FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY)
+#define FI_MR_SCALABLE		0
```

Comment:
Defining FI_MR_BASIC and FI_MR_SCALABLE allow the app to compile, but the behavior isn't the same.  FI_MR_BASIC, specifically will result is a broken app.  IMO, it's better to fail the compilation.

Reply:
Keep these temporarily for current CI testing. Will remove before merge.